pkg/store/image/image.go

```diff
@@ -45,6 +46,14 @@ type Image struct {
 	Image containerd.Image
 }
 
+func trimAlgorithmFromImageID(imageID string) string {
```

Comment:
Curious, how does trimming the prefix help?

Note: there is more than one valid prefix here to the left of the `:`.. so trimming it possibly looses context.

Reply:
Yes,it possibly looses context, but it only try. If try fail, it send the whole context to Parse.

Reply:
"Curious, how does trimming the prefix help?"
Becanse we need to support "sha256:xxx" and "xxx" as docker. we store "xxx" in truncindex.
So I must get "xxx" by trimming the prifix

Reply:
I see, so the decision/plan/design is to only store and retrieve the `xxx` portion of the image ref in the truncindex. That makes sense. Then we don't need to store it in both sha256:xxx and xxx.. trees. Sure ok that works. Just a small window where there could be two with different sha values, 256/512 and the same few id values. ok..

Reply:
yes, there is a small window.
This function is only for crictl. I think it is acceptable.

Reply:
>  Just a small window where there could be two with different sha values, 256/512 and the same few id values. And if that happens they should just get a false dup and have to use the full id.

Yeah. I agree. We should not remove the algorithm.

See my new comment.